### DISCUSSION TITLE
CodeQL currently unavailable?

### DISCUSSION THREAD

**Discussion by DGuhr** [github/codeql#13703] [Q&A] [answered]
Hey 👋 

We've just set up CodeQL on our repo with default mode. It all worked well until ~2h ago. All we did was switching the severity threshold from "high or higher" to "medium and higher". Afterwards I tried to manually trigger a run, which is stuck until now, see https://github.com/RedHatInsights/authz/actions/runs/5508341626 - also, suddenly on new PRs (see here: https://github.com/RedHatInsights/authz/pull/192 ) the CodeQL check is neither shown nor executed. 

Pls help me out here, did I do sth wrong?

Thanks!

**Comment by DGuhr** [comment]
forget it, for whatever reason codeql was disabled again - re-enabling helped.

**Comment by orhantoy** [comment, marked as answer]
Thanks for reporting this ❤️ We do not recommend retrying Actions when using CodeQL default mode/setup. We are looking into ways to reduce confusion with improved documentation and/or disabling retrying these types of *dynamic* workflows.

**Reply by DGuhr** [reply]
cool, thanks for the explanation. :)

**Reply by DGuhr** [reply]
@orhantoy sorry to ask again, but.. we now set up the advanced workflow. Here's the yaml file: https://github.com/RedHatInsights/authz/blob/main/.github/workflows/codeql.yml you got any idea why the settings still allow me to "set up" codeql (again)?  I darkly remember from another repository that they _should_ show the path to the codql yaml used instead, right? See screenshot
![Screenshot 2023-07-12 at 11 27 21](https://github.com/github/codeql/assets/89905860/0386818e-8f90-4dcf-8dc4-c2525c30ab65)

The CodeQL workflow is executed as defined in the yaml, though (I changed the usual name just to make sure), see e.g. https://github.com/RedHatInsights/authz/actions/runs/5529888149

**Reply by intrigus-lgtm** [reply]
I think you have to wait until the workflow has finished at least once.
Then the UI should show it as enabled.

**Reply by DGuhr** [reply]
thx @intrigus-lgtm - sadly that already happened a few times, see https://github.com/RedHatInsights/authz/actions - still it doesn't show up. But as long as it actually works, this is "just" a UI bug that may change when some caches are emptied later on, so I don't really worry.

**Reply by intrigus-lgtm** [reply]
Ah, then I don't know.
I had the problem I described recently so I thought you might have hit the same problem.

**Reply by orhantoy** [reply]
I can see how that is confusing but let me explain: Default Setup is still *available* so that's why you see that option. If you click on Set up > Default you will have the option to switch over to what we call the Default setup. In your case the dialog that pops up should also include a warning, saying that doing so would disable the workflow setup.
